lib/ansible/plugins/lookup/oc.py

```diff
@@ -0,0 +1,252 @@
+                if 'generated' not in resource['name']:
+                    self.kinds[resource['kind']] = \
+                        {'kind': resource['kind'],
+                         'name': resource['name'].split('/')[0],
+                         'namespaced': resource['namespaced'],
+                         'api': api,
+                         'version': 'v1',
+                         'baseurl': url
+                         }
+
+    def url(self, kind=None, namespace=None, resource_name=None, pretty=False, labelSelector=None, fieldSelector=None, resourceVersion=None):
+        firstParam = True
+        self.kind = kind
+
+        url = self.kinds[self.kind]['baseurl']
+        if self.kinds[self.kind]['namespaced'] is True:
+            url += '/namespaces/'
+            if namespace is None:
+                raise AnsibleError('Kind %s requires a namespace.  \
+                                    None provided' % self.kind)
+            url += namespace
+
+        url += '/'
+        url += self.kinds[self.kind]['name']
+
+        if resource_name is not None:
+            url += '/'
+            url += resource_name
+
+        if pretty:
+            url += '?pretty'
+            firstParam = False
+
+        if labelSelector is not None:
+            if firstParam:
+                url += '?'
+            else:
+                url += '&'
+
+            url += urlencode({'labelSelector': labelSelector})
+            firstParam = False
+
+        if fieldSelector is not None:
+            if firstParam:
+                url += '?'
+            else:
+                url += '&'
+
+            url += urlencode({'fieldSelector': fieldSelector})
+            firstParam = False
+
+        if resourceVersion is not None:
+            if firstParam:
+                url += '?'
+            else:
+                url += '&'
+
+            url += urlencode({'resourceVersion': resourceVersion})
+            firstParam = False
+
+        return url
+
+    def query(self, kind=None, namespace=None, resource_name=None, pretty=False, labelSelector=None, fieldSelector=None, resourceVersion=None):
+        url = self.url(kind=kind,
+                       namespace=namespace,
+                       resource_name=resource_name,
+                       pretty=pretty,
+                       labelSelector=labelSelector,
+                       fieldSelector=fieldSelector,
+                       resourceVersion=resourceVersion)
+
+        response = urls.open_url(url=url,
+                                 headers=self.headers,
+                                 validate_certs=self.validate_certs,
+                                 method='get')
+
+        if response.code >= 300:
+            raise AnsibleError("OC Query raised exception with code %s" +
+                               "and message %s against url %s" %
+                               str(response.code, response.read(), self.url()))
```

Comment:
I think this str is in the wrong place (and the string addition is unneeded there might be a bytes type that you want to convert on python3).  You probably need something more like:
``` python
raise AnsibleError("OC Query raised exception with code %s"
                   "and message %s against url %s" %
                   (response.code, to_native(response.read()), to_native(self.url()))
```

Reply:
Also, I think ```self.url()``` should be ```url```